app/src/main/java/org/jellyfin/androidtv/content/MediaContentProvider.kt

```diff
@@ -0,0 +1,171 @@
+package org.jellyfin.androidtv.content
+
+import android.app.SearchManager
+import android.content.ContentProvider
+import android.content.ContentValues
+import android.content.Intent
+import android.content.UriMatcher
+import android.database.Cursor
+import android.database.MatrixCursor
+import android.net.Uri
+import android.provider.BaseColumns
+import io.ktor.util.*
+import kotlinx.coroutines.*
+import org.jellyfin.androidtv.TvApp
+import org.jellyfin.androidtv.di.systemApiClient
+import org.jellyfin.apiclient.model.querying.ItemSortBy
+import org.jellyfin.sdk.api.client.KtorClient
+import org.jellyfin.sdk.api.operations.ImageApi
+import org.jellyfin.sdk.api.operations.ItemsApi
+import org.jellyfin.sdk.model.api.*
+import org.jellyfin.sdk.model.serializer.toUUIDOrNull
+import org.koin.core.component.KoinApiExtension
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
+import timber.log.Timber
+import java.text.SimpleDateFormat
+
+
+@KoinApiExtension
+class MediaContentProvider : ContentProvider(), KoinComponent {
+
+	companion object {
+		const val AUTHORITY = "org.jellyfin.androidtv.content"
+		const val SUGGEST_PATH = "suggestions"
+		private const val TICKS_IN_MILLISECOND = 10000
+	}
+
+	private val apiClient by inject<KtorClient>(systemApiClient)
+	private val itemsApi by lazy { ItemsApi(apiClient) }
+	private val imageApi by lazy { ImageApi(apiClient) }
+
+	private val SEARCH_SUGGEST = 1
+
+	private val sURIMatcher = UriMatcher(UriMatcher.NO_MATCH)
+
+	init {
+		sURIMatcher.addURI(AUTHORITY, SUGGEST_PATH + "/" + SearchManager.SUGGEST_URI_PATH_QUERY, SEARCH_SUGGEST)
+		sURIMatcher.addURI(AUTHORITY, SUGGEST_PATH + "/" + SearchManager.SUGGEST_URI_PATH_QUERY + "/*", SEARCH_SUGGEST)
+	}
```

Comment:
```suggestion
	private val sURIMatcher = UriMatcher(UriMatcher.NO_MATCH).apply {
		addURI(AUTHORITY, SUGGEST_PATH + "/" + SearchManager.SUGGEST_URI_PATH_QUERY, SEARCH_SUGGEST)
		addURI(AUTHORITY, SUGGEST_PATH + "/" + SearchManager.SUGGEST_URI_PATH_QUERY + "/*", SEARCH_SUGGEST)
	}
```

Can also be moved to companion object